### PR TITLE
backup: hash backup file name

### PR DESCRIPTION
**Pull request by 3pointer**
Signed-off-by: luancheng <luancheng@pingcap.com>

###  What have you changed?
use start key sha256 to replace raw start key, if we use start key as file name we could meet 
"file name too long" error because some index key are longer than 255 bytes

###  What is the type of the changes?

- Bugfix (a change which fixes an issue)

###  How is the PR tested?
Please select the tests that you ran to verify your changes:
- Unit test
- Integration test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?
N/A

###  Does this PR affect `tidb-ansible`?
N/A

### CODE REVIEW

File: components/backup/src/endpoint.rs [hunk 1]
```diff
@@ -568,10 +568,10 @@ fn get_max_start_key(start_key: Option<&Key>, region: &Region) -> Option<Key> {
...
 /// A name consists with three parts: store id, region_id and a epoch version.
-fn backup_file_name(store_id: u64, region: &Region, key: Option<String>) -> String {
+fn backup_file_name(store_id: u64, region: &Region, key: Option<Vec<u8>>) -> String {
     match key {
         Some(k) => format!(
-            "{}_{}_{}_{}",
+            "{}_{}_{}_{:?}",
             store_id,
             region.get_id(),
             region.get_region_epoch().get_version(),
```
Comment (overvenus) [inline components/backup/src/endpoint.rs]: Should use `"{}", hex::encode(k)`.

Here is an example of `{:?}, vec![1, 2, 3]` => `[1, 2, 3] `

File: components/backup/src/endpoint.rs [hunk 2]
```diff
@@ -382,7 +382,7 @@ impl<E: Engine, R: RegionInfoProvider> Endpoint<E, R> {
                 let key = brange
                     .start_key
                     .clone()
-                    .map(|k| hex::encode(k.into_raw().unwrap()));
+                    .and_then(|k| tikv_util::file::sha256(&k.into_raw().unwrap()).ok());
```
Comment (overvenus) [inline components/backup/src/endpoint.rs]: Could you add some comments about why we need it?

Comment (3pointer) [reply components/backup/src/endpoint.rs]: sure, add [comment](https://github.com/tikv/tikv/pull/6177/files#diff-53831868d19a8c3495a300c681e252a8R383)

File: components/backup/src/endpoint.rs (hunk 1 above)
Comment (3pointer) [reply components/backup/src/endpoint.rs]: `hex::encode` is used in [sha256](https://github.com/tikv/tikv/blob/64b905954c1f4fdf00c10e3f64932240b0843d13/components/tikv_util/src/file.rs#L85, I think use `String::from_utf8` is enough

**Pull request merged**